### PR TITLE
Update Maven to 3.8.6, maven-wrapper to 3.1.1 and maven-invoker to 3.2.0

### PR DESCRIPTION
**Pull request by famod**
https://issues.apache.org/jira/secure/ReleaseNote.jspa?projectId=12316922&version=12351556

> [[MNG-7432](https://issues.apache.org/jira/browse/MNG-7432)] - [REGRESSION] Resolver session contains non-MavenWorkspaceReader

is the one that caused Quarkus to _not_ update to 3.8.5.

PS: Bumping wrapper and invoker is not mandatory, but I saw those were outdated and just updated them as well.

### CODE REVIEW

File: independent-projects/bootstrap/maven-resolver/src/main/java/io/quarkus/bootstrap/resolver/maven/BootstrapMavenContext.java [hunk 1]
```diff
@@ -968,19 +968,17 @@ private static FileProfileActivator createFileProfileActivator() throws Bootstra
...
         var translator = new DefaultPathTranslator();
         try {
-            activator.setPathTranslator(translator);
-        } catch (LinkageError e) {
-            // setPathTranslator() not found; Maven >= 3.8.5 (https://github.com/apache/maven/pull/649)
+            var pathInterpolator = new ProfileActivationFilePathInterpolator();
+            pathInterpolator.setPathTranslator(translator);
+            activator.setProfileActivationFilePathInterpolator(pathInterpolator);
+        } catch (NoClassDefFoundError e) {
+            // ProfileActivationFilePathInterpolator not found; Maven < 3.8.5 (https://github.com/apache/maven/pull/649)
```
Comment (famod) [inline independent-projects/bootstrap/maven-resolver/src/main/java/io/quarkus/bootstrap/resolver/maven/BootstrapMavenContext.java:975]: Re-tested this via #24643

File: test-framework/maven/src/main/java/io/quarkus/maven/it/verifier/MavenProcessInvoker.java [hunk 2]
```diff
@@ -48,7 +48,7 @@ public InvocationResult execute(InvocationRequest request) throws MavenInvocatio
 
         File workingDirectory = getWorkingDirectory();
         if (workingDirectory != null) {
-            cliBuilder.setWorkingDirectory(getWorkingDirectory());
+            cliBuilder.setBaseDirectory(getWorkingDirectory());
```
Comment (famod) [inline test-framework/maven/src/main/java/io/quarkus/maven/it/verifier/MavenProcessInvoker.java:51]: See https://github.com/apache/maven-invoker/commit/ca84e5c6108366b800c2e61d377eadcb327d6499#diff-7a2ee223cf0ee38d98ed325df11438abe1b5134ce5d7430933461efc0ff5963cL725
I think this change should be ok given the way Quarkus is using this method.

**Comment by famod** [comment]
Good candidate for 2.10.0 AFAICS, but too bad it comes too late for CR1.

File: test-framework/maven/src/main/java/io/quarkus/maven/it/verifier/MavenProcessInvoker.java (hunk 2 above)
Comment (famod) [reply test-framework/maven/src/main/java/io/quarkus/maven/it/verifier/MavenProcessInvoker.java:51]: This is a change in `maven-invoker`, not in actual Maven, so I don't think Maven < 3.8.6 will have a problem with this.

Should I break this up in three PRs?

**Review by aloubyansky — APPROVED** [review]
Thanks @famod

**Comment by famod** [comment]
@gsmet any objections?

**Comment by famod** [comment]
I'll go ahead and merge this.

**Comment by rsvoboda** [comment]
I think this PR should have noteworthy (or breaking change ) label. And probably migration guide entry.

`maven-wagon` was updated to 3.5.1 and no longer pulls in `jsoup` dependency.

We have seen two kinds of troubles

A) failures with consul extension - https://github.com/quarkiverse/quarkus-config-extensions/issues/70

B) usage of jsoup in tests, jsoup was transitive dependency of `io.quarkus:quarkus-junit5` and people probably didn't define jsoup explicitly in their projects

**Comment by gsmet** [comment]
Hmmm, let's not backport it then.

**Comment by gsmet** [comment]
I think we shouldn't. JSoup sometimes have CVE so I would rather avoid it if we don't have a dependency on it.

**Comment by famod** [comment]
So the idea is to:
- remove jsoup from `independent-projects/bootstrap/**`
- add it to `build-parent`
- revisit and probably remove the exclusion from docs

❓

**Comment by gsmet** [comment]
If we are not depending on it, yes. Now I'm a bit unclear if we are still dragging it in some cases?

**Comment by aloubyansky** [comment]
The extension processor uses it. I would say yes to your plan @famod. Were you planning to take care of that?

**Comment by rsvoboda** [comment]
I created https://github.com/quarkusio/quarkus/issues/26141 for jsoup in the morning, these discussions should be probably moved there

**Comment by rsvoboda** [comment]
about `add it to build-parent (just managed)` .. there was https://github.com/quarkusio/quarkus/pull/21406 for move from build-parent to bootstrap-bom in November last year ;)

**Comment by aloubyansky** [comment]
True, the bootstrap doesn't use it any more though.

**Pull request merged**